databuilder/databuilder/publisher/neo4j_csv_unwind_publisher.py

```diff
@@ -0,0 +1,501 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import csv
+import ctypes
+import logging
+import time
+from io import open
+from os import listdir
+from os.path import isfile, join
+from typing import (
+    Dict, List, Set,
+)
+
+import neo4j
+import pandas
+from jinja2 import Template
+from neo4j import GraphDatabase, Transaction
+from neo4j.api import (
+    SECURITY_TYPE_SECURE, SECURITY_TYPE_SELF_SIGNED_CERTIFICATE, parse_neo4j_uri,
+)
+from neo4j.exceptions import Neo4jError
+from pyhocon import ConfigFactory, ConfigTree
+
+from databuilder.publisher.base_publisher import Publisher
+
+# Setting field_size_limit to solve the error below
+# _csv.Error: field larger than field limit (131072)
+# https://stackoverflow.com/a/54517228/5972935
+csv.field_size_limit(int(ctypes.c_ulong(-1).value // 2))
+
+# Config keys
+# A directory that contains CSV files for nodes
+NODE_FILES_DIR = 'node_files_directory'
+# A directory that contains CSV files for relationships
+RELATION_FILES_DIR = 'relation_files_directory'
+# A end point for Neo4j e.g: bolt://localhost:9999
+NEO4J_END_POINT_KEY = 'neo4j_endpoint'
+# A transaction size that determines how often it commits.
+NEO4J_TRANSACTION_SIZE = 'neo4j_transaction_size'
+# A boolean flag to make it fail if relationship is not created
+NEO4J_RELATIONSHIP_CREATION_CONFIRM = 'neo4j_relationship_creation_confirm'
+
+NEO4J_MAX_CONN_LIFE_TIME_SEC = 'neo4j_max_conn_life_time_sec'
+
+# list of nodes that are create only, and not updated if match exists
+NEO4J_CREATE_ONLY_NODES = 'neo4j_create_only_nodes'
+
+NEO4J_USER = 'neo4j_user'
+NEO4J_PASSWORD = 'neo4j_password'
+# in Neo4j (v4.0+), we can create and use more than one active database at the same time
+NEO4J_DATABASE_NAME = 'neo4j_database'
+
+# NEO4J_ENCRYPTED is a boolean indicating whether to use SSL/TLS when connecting
+NEO4J_ENCRYPTED = 'neo4j_encrypted'
+# NEO4J_VALIDATE_SSL is a boolean indicating whether to validate the server's SSL/TLS
+# cert against system CAs
+NEO4J_VALIDATE_SSL = 'neo4j_validate_ssl'
+
+# This will be used to provide unique tag to the node and relationship
+JOB_PUBLISH_TAG = 'job_publish_tag'
+
+# any additional fields that should be added to nodes and rels through config
+ADDITIONAL_FIELDS = 'additional_fields'
+
+# Neo4j property name for published tag
+PUBLISHED_TAG_PROPERTY_NAME = 'published_tag'
+
+# Neo4j property name for last updated timestamp
+LAST_UPDATED_EPOCH_MS = 'publisher_last_updated_epoch_ms'
+
+# A boolean flag to indicate if publisher_metadata (e.g. published_tag,
+# publisher_last_updated_epoch_ms)
+# will be included as properties of the Neo4j nodes
+ADD_PUBLISHER_METADATA = 'add_publisher_metadata'
+
+# A boolean to indicate whether to publish reverse relationships, otherwise there will only be one direction
+PUBLISH_REVERSE_RELATIONSHIPS = 'publish_reverse_relationships'
```

Comment:
let's make sure to note that this is a super secret option which should not be used generally